### PR TITLE
Amls 4884 - Risk assessment split

### PR DESCRIPTION
**Pull request by matthewharris1**
AMLS-4884 - Risk assessment split in business activities. 

## How Has This Been Tested?

Unit Testing
Manual Testing

## Checklist

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [x] Requires acceptance test run.
- [x] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

File: app/controllers/businessactivities/RiskAssessmentController.scala [hunk 1]
```diff
@@ -59,21 +59,21 @@ class RiskAssessmentController @Inject() (val dataCacheConnector: DataCacheConne
...
         }
       }
   }
 
-  private def redirectDependingOnEdit(edit: Boolean, accountancyServices: Boolean) = edit match {
-    case true => Redirect(routes.SummaryController.get())
-    case false => accountancyServices match {
-      case true => Redirect(routes.SummaryController.get())
-      case false => Redirect(routes.AccountantForAMLSRegulationsController.get())
+  private def redirectDependingOnEdit(edit: Boolean, accountancyServices: Boolean, data: RiskAssessmentHasPolicy) =
```
Comment (slavvco) [inline app/controllers/businessactivities/RiskAssessmentController.scala]: here might be necessary some refactor, first and third case for example.

Comment (matthewharris1) [reply app/controllers/businessactivities/RiskAssessmentController.scala]: Done

Comment (matthewharris1) [reply app/controllers/businessactivities/RiskAssessmentController.scala]: Removed 3rd statement.

File: app/controllers/businessactivities/DocumentRiskAssessmentController.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            } yield redirectDependingOnEdit(edit, ControllerHelper.isAccountancyServicesSelected(Some(businessMatching)))
+
+          }
+        } recoverWith {
+          case _: IndexOutOfBoundsException => Future.successful(NotFound(notFoundView))
+        }
+      }
+  }
+
+  private def redirectDependingOnEdit(edit: Boolean, accountancyServices: Boolean) = (edit, accountancyServices) match {
```
Comment (drewmiley) [inline app/controllers/businessactivities/DocumentRiskAssessmentController.scala]: Could just use if/else?

File: app/controllers/businessactivities/RiskAssessmentController.scala [hunk 3]
```diff
@@ -59,21 +59,19 @@ class RiskAssessmentController @Inject() (val dataCacheConnector: DataCacheConne
...
           case _: IndexOutOfBoundsException => Future.successful(NotFound(notFoundView))
         }
       }
   }
 
-  private def redirectDependingOnEdit(edit: Boolean, accountancyServices: Boolean) = edit match {
-    case true => Redirect(routes.SummaryController.get())
-    case false => accountancyServices match {
+  private def redirectDependingOnAccountancyServices(accountancyServices: Boolean, data: RiskAssessmentHasPolicy) =
+    accountancyServices match {
```
Comment (drewmiley) [inline app/controllers/businessactivities/RiskAssessmentController.scala]: Match on both?

File: app/models/businessactivities/BusinessActivities.scala [hunk 4]
```diff
@@ -91,6 +91,26 @@ case class BusinessActivities(
     this.copy(riskAssessmentPolicy = Some(p), hasChanged = hasChanged || !this.riskAssessmentPolicy.contains(p),
       hasAccepted = hasAccepted && this.riskAssessmentPolicy.contains(p))
 
+  def riskAssessmentHasPolicy(p: RiskAssessmentHasPolicy): BusinessActivities = {
+    val newRiskAssessmentPolicy = if (this.riskAssessmentPolicy.isDefined) {
```
Comment (drewmiley) [inline app/models/businessactivities/BusinessActivities.scala]: Use map

File: app/models/businessactivities/BusinessActivities.scala [hunk 5]
```diff
@@ -91,6 +91,26 @@ case class BusinessActivities(
...
+      this.riskAssessmentPolicy.get.copy(hasPolicy = p)
+    } else {
+      RiskAssessmentPolicy(p, RiskAssessmentTypes(Set()))
+    }
+    this.copy(riskAssessmentPolicy = Some(newRiskAssessmentPolicy), hasChanged = hasChanged || !this.riskAssessmentPolicy.contains(p),
+      hasAccepted = hasAccepted && this.riskAssessmentPolicy.contains(p))
+  }
+
+  def riskAssessmentTypes(p: RiskAssessmentTypes): BusinessActivities = {
+    val newRiskAssessmentPolicy = if (riskAssessmentPolicy.isDefined) {
```
Comment (drewmiley) [inline app/models/businessactivities/BusinessActivities.scala]: Use map

File: app/models/businessactivities/RiskAssessmentHasPolicy.scala [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+  implicit val formRule: Rule[UrlFormEncoded, RiskAssessmentHasPolicy] = From[UrlFormEncoded] { __ =>
+    import jto.validation.forms.Rules._
+    (__ \ "hasPolicy").read[Boolean].withMessage("error.required.ba.option.risk.assessment") flatMap {
+      RiskAssessmentHasPolicy.apply
+    }
+  }
+
+  implicit def formWrites: Write[RiskAssessmentHasPolicy, UrlFormEncoded] = Write {
+    case RiskAssessmentHasPolicy(hasPolicy) => Map("hasPolicy" -> Seq(hasPolicy.toString))
```
Comment (drewmiley) [inline app/models/businessactivities/RiskAssessmentHasPolicy.scala]: Seq?

File: app/models/businessactivities/RiskAssessmentPolicy.scala [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+case class RiskAssessmentPolicy(hasPolicy: RiskAssessmentHasPolicy, riskassessments: RiskAssessmentTypes)
+
+object RiskAssessmentPolicy {
+
+  implicit val jsonReads: Reads[RiskAssessmentPolicy] = {
+    import play.api.libs.functional.syntax._
+    ((__ \ "hasPolicy").read[Boolean] and
+      (__ \ "riskassessments").readNullable[Set[RiskAssessmentType]]
+      )((a, b) => RiskAssessmentPolicy(RiskAssessmentHasPolicy(a), RiskAssessmentTypes(b.getOrElse(Set[RiskAssessmentType]()))))
```
Comment (drewmiley) [inline app/models/businessactivities/RiskAssessmentPolicy.scala]: Name variables more clearly

File: app/models/businessactivities/RiskAssessmentPolicy.scala [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  implicit val jsonWrites:Writes[RiskAssessmentPolicy] = {
+    Writes[RiskAssessmentPolicy] {
+      case RiskAssessmentPolicy(RiskAssessmentHasPolicy(true), RiskAssessmentTypes(a)) =>
+        Json.obj(
+          "hasPolicy" -> true,
+          "riskassessments" -> a.toSeq.map(d =>
```
Comment (drewmiley) [inline app/models/businessactivities/RiskAssessmentPolicy.scala]: Name variables more clearly

File: app/models/businessactivities/RiskAssessmentPolicy.scala [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+          d.toString match {
```
Comment (drewmiley) [inline app/models/businessactivities/RiskAssessmentPolicy.scala]: Indent

File: app/models/businessactivities/RiskAssessmentPolicy.scala [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+            case "PaperBased" => "01"
+            case "Digital" => "02"
+            case _ => ""
```
Comment (drewmiley) [inline app/models/businessactivities/RiskAssessmentPolicy.scala]: Is it possible to remove this line? If so, we should

File: app/views/businessactivities/document_risk_assessment_policy.scala.html [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    @form(f, controllers.businessactivities.routes.DocumentRiskAssessmentController.post(edit)) {
+
+        @fieldsetWithErrorGroup(
+            f = f("riskassessments"),
+            heading = "businessactivities.document.riskassessment.policy.title",
+            section = "summary.businessactivities",
+            hint = "lbl.select.all",
+            panel = false
+        ) {
+            <input type="hidden" name="hasPolicy" value=true />
```
Comment (drewmiley) [inline app/views/businessactivities/document_risk_assessment_policy.scala.html]: Not sure this is required anymore

File: app/views/businessactivities/document_risk_assessment_policy.scala.html [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+            @checkbox(
+                f = f("riskassessments[]"),
+                labelText = "businessactivities.RiskAssessmentType.lbl.01",
```
Comment (drewmiley) [inline app/views/businessactivities/document_risk_assessment_policy.scala.html]: I don't like the opposition of the 01 and 02 between the labels and the values

File: app/controllers/businessactivities/RiskAssessmentController.scala (hunk 3 above)
Comment (matthewharris1) [reply app/controllers/businessactivities/RiskAssessmentController.scala]: Swav said to just use data as an if case as its only used once. Guess its down to preference?

**Pull request merged**